conformance/tests/enums_expansion.py

```diff
@@ -0,0 +1,78 @@
+"""
+Tests that the type checker handles literal expansion of enum classes.
+"""
+
+# Specification: https://typing.readthedocs.io/en/latest/spec/enums.html#enum-literal-expansion
+
+from enum import Enum, Flag
+from typing import Literal, Never, assert_type
+
+# > From the perspective of the type system, most enum classes are equivalent
+# > to the union of the literal members within that enum. Type checkers may
+# > therefore expand an enum type
+
+
+class Color(Enum):
+    RED = 1
+    GREEN = 2
+    BLUE = 3
+
+
+def print_color1(c: Color):
+    if c is Color.RED or c is Color.BLUE:
+        print("red or blue")
+    else:
+        assert_type(c, Literal[Color.GREEN])  # E?
+
+
+def print_color2(c: Color):
+    match c:
+        case Color.RED | Color.BLUE:
+            print("red or blue")
+        case Color.GREEN:
+            print("green")
+        case _:
+            assert_type(c, Never)  # E?
+
+
+# > This rule does not apply to classes that derive from enum.Flag because
+# > these enums allow flags to be combined in arbitrary ways.)
+
+
+class CustomFlags(Flag):
+    FLAG1 = 1
+    FLAG2 = 2
+    FLAG3 = 4
+
+
+def test1(f: CustomFlags):
+    if f is CustomFlags.FLAG1 | CustomFlags.FLAG2:
```

Comment:
```suggestion
    if f is CustomFlags.FLAG1 or f is CustomFlags.FLAG2:
```
I feel this makes the test a bit more interesting, since now a type checker that implements Flags naively should actually fail the test.

Reply:
Oh, but apparently mypy does perform type narrowing on the code as you wrote it? That's bad.